src/Query.tsx

```diff
@@ -201,9 +215,14 @@ class Query<
   render() {
     const { children } = this.props;
     const queryResult = this.getQueryResult();
-    return children(queryResult);
+    return children(this.shouldSkip() ? undefined : queryResult);
   }
 
+  private shouldSkip = (props = this.props) => {
```

Comment:
instead of a function, this we should take `skip` as a boolean prop which makes it easy to check against previous props and lessen the internal function calls.